diff --git a/compiler/options.nim b/compiler/options.nim
index 6372cdd..1354da0 100644
--- a/compiler/options.nim
+++ b/compiler/options.nim
@@ -229,6 +229,10 @@ proc setDefaultLibpath*() =
       else: libpath = joinPath(prefix, "lib")
     else: libpath = joinPath(prefix, "lib")
 
+    if not fileExists(libpath / "system.nim") and
+        fileExists(libpath / "nim" / "system.nim"):
+      libpath = libpath / "nim"
+
     # Special rule to support other tools (nimble) which import the compiler
     # modules and make use of them.
     let realNimPath = findExe("nim")
