--- a/lib/wrappers/openssl.nim
+++ b/lib/wrappers/openssl.nim
@@ -53,7 +53,7 @@
     from posix import SocketHandle
 
 elif useWinVersion:
-  when not defined(nimOldDlls) and defined(cpu64):
+  when defined(cpu64):
     const
       DLLSSLName* = "(libssl-1_1-x64|ssleay64|libssl64).dll"
       DLLUtilName* = "(libcrypto-1_1-x64|libeay64).dll"
--- a/lib/wrappers/pcre.nim
+++ b/lib/wrappers/pcre.nim
@@ -314,12 +314,7 @@
 
 when not defined(usePcreHeader):
   when hostOS == "windows":
-    when defined(nimOldDlls):
-      const pcreDll = "pcre.dll"
-    elif defined(cpu64):
-      const pcreDll = "pcre64.dll"
-    else:
-      const pcreDll = "pcre32.dll"
+    const pcreDll = "libpcre-1.dll"
   elif hostOS == "macosx":
     const pcreDll = "libpcre(.3|.1|).dylib"
   else:
--- a/lib/wrappers/sqlite3.nim
+++ b/lib/wrappers/sqlite3.nim
@@ -13,12 +13,7 @@
   {.push styleChecks: off.}
 
 when defined(windows):
-  when defined(nimOldDlls):
-    const Lib = "sqlite3.dll"
-  elif defined(cpu64):
-    const Lib = "sqlite3_64.dll"
-  else:
-    const Lib = "sqlite3_32.dll"
+  const Lib = "libsqlite3-0.dll"
 elif defined(macosx):
   const
     Lib = "libsqlite3(|.0).dylib"
